### PR TITLE
MODDICORE-181: Create mapper to generate Authority objects

### PR DESCRIPTION
**Pull request by Shans-Kaluhin**
### Purpose
 To have the ability to generate the Authority domain object by the incoming Marc record using default mapping rules

### Approach
  - Create MarcToAuthorityMapper. It should work similarly to MarcToHoldingsMapper using the Processor
  - Temporarily remove 001 rule

###  Learn
[MODDICORE-181](https://issues.folio.org/browse/MODDICORE-181)

### CODE REVIEW

File: src/test/resources/org/folio/processing/mapping/mappedAuthorityRecord.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+  "id": "0",
```
Comment (psmagin) [inline src/test/resources/org/folio/processing/mapping/mappedAuthorityRecord.json:2]: Why the id is 0. It should be an uuid from 999 field: b90cb1bc-601f-45d7-b99e-b11efd281dcd

**Comment by Shans-Kaluhin** [comment]
Hello @psmagin,
After discussion with @Igor-Gorchakov, we decide to remove `001` for now.
Because Processor does not support mapping from Marc control field to the composite field.
After adding this functionality, we will return `001` rule for modules 

See [MODDICORE-197](https://issues.folio.org/browse/MODDICORE-197)

File: src/test/java/org/folio/processing/mapping/AuthorityMappingTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    "src/test/resources/org/folio/processing/mapping/authorityRules.json";
+
+  private final RecordMapper<Authority> mapper = RecordMapperBuilder.buildMapper("MARC_AUTHORITY");
+
+  @Test
+  public void testMarcToAuthority() throws IOException {
+    JsonObject expectedMappedHoldings = new JsonObject(TestUtil.readFileFromPath(MAPPED_AUTHORITY_PATH));
+    JsonObject mappingRules = new JsonObject(TestUtil.readFileFromPath(DEFAULT_MAPPING_RULES_PATH));
+
+    Authority actualMappedHoldings = mapper.mapRecord(getJsonMarcRecord(), new MappingParameters(), mappingRules);
```
Comment (psmagin) [inline src/test/java/org/folio/processing/mapping/AuthorityMappingTest.java:41]:
```suggestion
    Authority actualMappedAuthority = mapper.mapRecord(getJsonMarcRecord(), new MappingParameters(), mappingRules);
```

File: src/test/java/org/folio/processing/mapping/AuthorityMappingTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  private static final String MAPPED_AUTHORITY_PATH =
+    "src/test/resources/org/folio/processing/mapping/mappedAuthorityRecord.json";
+  private static final String DEFAULT_MAPPING_RULES_PATH =
+    "src/test/resources/org/folio/processing/mapping/authorityRules.json";
+
+  private final RecordMapper<Authority> mapper = RecordMapperBuilder.buildMapper("MARC_AUTHORITY");
+
+  @Test
+  public void testMarcToAuthority() throws IOException {
+    JsonObject expectedMappedHoldings = new JsonObject(TestUtil.readFileFromPath(MAPPED_AUTHORITY_PATH));
```
Comment (psmagin) [inline src/test/java/org/folio/processing/mapping/AuthorityMappingTest.java:38]:
```suggestion
    JsonObject expectedMappedAuthority = new JsonObject(TestUtil.readFileFromPath(MAPPED_AUTHORITY_PATH));
```

File: src/test/java/org/folio/processing/mapping/AuthorityMappingTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    JsonObject mappingRules = new JsonObject(TestUtil.readFileFromPath(DEFAULT_MAPPING_RULES_PATH));
+
+    Authority actualMappedHoldings = mapper.mapRecord(getJsonMarcRecord(), new MappingParameters(), mappingRules);
+    Assert.assertEquals(JsonObject.mapFrom(actualMappedHoldings).put("id", "0").encode(), expectedMappedHoldings.encode());
```
Comment (psmagin) [inline src/test/java/org/folio/processing/mapping/AuthorityMappingTest.java:42]: Remove setting of id to 0 and check that mapping rule for 999 field is working

File: src/test/java/org/folio/processing/mapping/AuthorityMappingTest.java (hunk 2 above)
Comment (Shans-Kaluhin) [reply src/test/java/org/folio/processing/mapping/AuthorityMappingTest.java]: My bad 😅

**Pull request merged**